Document Collectibles (#450)
discord-userdoccers/discord-userdoccers: pages/resources/collectibles.mdx

| themes?         | array[string] | The products with the specified [theme](#shop-product-themes) to search for |
| orbs_eligible?  | boolean       | Whether you can purchase the products with orbs                             |
| offset?         | integer       | Number of products to skip before returning results                         |
| limit?          | integer       | Max amount of SKU IDs to return (max 100, default 20)                       |
| sort_type?      | string        | How should the results be [sorted](#shop-sort-type)                         |
| sort_direction? | string        | The direction to sort the results in (`asc` or `desc`, default `desc`)      |
| search?         | string        | The query to match (max 256 characters)                                     |

###### Shop Item Type

| Value             | Description                      |
| ----------------- | -------------------------------- |
| AVATAR_DECORATION | An avatar decoration product     |
| PROFILE_EFFECT    | A profile effect product         |
| NAMEPLATE         | A nameplate product              |
| BUNDLE            | A bundle of collectible products |

> ###### Shop Product Themes

Reviewer: ```suggestion
###### Shop Product Theme
```

| category_sku_id        | snowflake                                                                                  | The category SKU ID of the collectible product                                                               |
| expires_at? ^2^        | ?ISO8601 timestamp                                                                         | The time at which the collectible product expires                                                            |
| google_sku_ids?        | map[integer, string]                                                                       | The Google SKU IDs for the collectible per [purchase type](/resources/store#subscription-plan-purchase-type) |
| items ^3^              | array[[collectible item](#collectible-item-object) object]                                 | The items included in the collectible product                                                                |
| name                   | string                                                                                     | The name of the collectible product                                                                          |

> | name_localizations?    | ?map[string, string]                                                                       | The localized names of the collectible product                                                               |

Reviewer: ```suggestion
```
internal implementation detail leak

| variants? ^1^          | array[collectible product](#collectible-product-structure) object]                         | The variants of the collectible product                                                                      |

^1^ Only included when `variant_return_style` is set to `VARIANTS_GROUP`, otherwise each variant will be returned as a separate product.

^2^ Only present on collectibles purchases endpoint.

^3^ If the product type is `VARIANTS_GROUP`, `items` will be empty and instead `variants` should be used.

###### Collectible Product Type

| Value | Name              | Description                            |
| ----- | ----------------- | -------------------------------------- |
| 0     | AVATAR_DECORATION | An avatar decoration                   |
| 1     | PROFILE_EFFECT    | A profile effect                       |
| 2     | NAMEPLATE         | A nameplate                            |
| 100   | NONE              | No collectible product                 |
| 1000  | BUNDLE            | A bundle of collectibles               |
| 2000  | VARIANTS_GROUP    | A group of variants                    |

> | 3000  | EXTERNAL_SKU      | An external SKU (e.g Fractional Nitro) |

Reviewer: ```suggestion
| 3000  | EXTERNAL_SKU      | A non-collectible SKU (e.g. fractional premium) |
```

###### Collectible Purchase Type

| Value | Name         | Description                                   |
| ----- | ------------ | --------------------------------------------- |

> | 1     | PURCHASED    | Product purchased using orbs or fiat currency |

Reviewer: ```suggestion
| 1     | PURCHASED    | Purchased normally |
```

| 5     | PROMOTIONAL  | Received through a promotional event          |
| 6     | GIFTED       | Received as a gift                            |
| 7     | SUBSCRIPTION | Claimed with a Nitro subscription             |
| 10    | QUEST        | Received from a quest                         |

> ### Collectible Category Object

Reviewer: move below collectible item

###### Profile Effect Position Structure

| Field | Type    | Description                             |
| ----- | ------- | --------------------------------------- |
| x     | integer | The x-coordinate of the animation frame |
| y     | integer | The y-coordinate of the animation frame |

###### Profile Effect Source Structure

| Field | Type   | Description                                     |
| ----- | ------ | ----------------------------------------------- |
| src   | string | The URL of the animation image (in APNG format) |

###### Profile Effect Animation Type

| Value | Name         | Description                        |
| ----- | ------------ | ---------------------------------- |
| 0     | UNSPECIFIED  | The animation type is unspecified  |
| 1     | PERSISTENT   | The animation type is persistent   |
| 2     | INTERMITTENT | The animation type is intermittent |

###### Nameplate Collectible Item Structure

| Field   | Type                                         | Description                                                  |
| ------- | -------------------------------------------- | ------------------------------------------------------------ |

> | type    | integer                                      | The [collectible product type](#collectible-product-type)    |

Reviewer: ```suggestion
| type    | integer                                      | The [type of collectible](#collectible-product-type)    |
```
consistency

| assets  | [item assets](#item-assets-structure) object | The URLs for the static and animated images of the nameplate |
| label   | string                                       | The nameplate accessibility description                      |

###### Nameplate Color Palette

| Value      | Name      |
| ---------- | --------- |
| none       | None      |
| crimson    | Crimson   |
| berry      | Berry     |
| sky        | Sky       |
| teal       | Teal      |
| forest     | Forest    |
| bubble_gum | BubbleGum |
| violet     | Violet    |
| cobalt     | Cobalt    |
| clover     | Clover    |
| lemon      | Lemon     |
| white      | White     |

###### Item Assets Structure

| Field              | Type   | Description                                                        |
| ------------------ | ------ | ------------------------------------------------------------------ |
| static_image_url   | string | The URL for the static image of the collectible                    |
| animated_image_url | string | The URL for the animated image of the collectible (in APNG format) |
| video_url?         | string | The URL for the video of the collectible                           |

> ### Shop Block Object

Reviewer: same nit as the above tagged union. move shop block type to the top and link each structure in the name field.
also make sure `type` is the first field in each structure for clarity.

###### Hero Shop Block Structure

| Field                     | Type                                           | Description                                                                                     |
| ------------------------- | ---------------------------------------------- | ----------------------------------------------------------------------------------------------- |
| banner_asset              | [asset config](#asset-config-structure) object | The banner asset                                                                                |
| banner_display_config     | [banner asset](#banner-asset-structure) object | The display configuration                                                                       |
| banner_text_color?        | string                                         | The hex color code for the banner text                                                          |

> | category_sku_id           | snowflake                                      | The [collectible category](#collectible-category-structure) SKU ID                              |

Reviewer: ```suggestion
| category_sku_id           | snowflake                                      | The collectible category SKU ID                              |
```
unnecessary to link object when it isn't referenced (apply to rest of fields too)

| hero_banner_animated_url  | string                                         | The URL of the animated hero banner image                                                       |
| hero_banner_url           | string                                         | The URL of the hero banner image                                                                |
| hero_logo_url             | string                                         | The URL of the hero logo image                                                                  |
| hero_rive_url?            | string                                         | The URL of the hero [Rive](https://rive.app/docs/runtimes/advanced-topic/format) animation      |
| logo_display_config       | [banner asset](#banner-asset-structure) object | The shop block logo display config                                                              |
| logo_url                  | string                                         | The URL of the hero logo image                                                                  |

> | name                      | string                                         | Name of the shop block                                                                          |

Reviewer: ```suggestion
| name                      | string                                         | The name of the shop block                                                                          |
```

| type                | integer                                        | The [shop block type](#shop-block-type)        |
| title               | string                                         | The title of the immersive banner              |
| body                | string                                         | The body text of the immersive banner          |
| help_center_url     | string                                         | The URL to the help center article             |
| text_color          | string                                         | The hex color code of the text                 |
| end_time            | ?ISO8601 timestamp                             | The end time of the immersive banner           |
| banner_asset        | [asset config](#asset-config-structure) object | The banner asset                               |
| banner_url          | string                                         | The URL of the immersive banner image          |
| banner_animated_url | string                                         | The URL of the animated immersive banner image |

> ### Collectibles Marketing Object

Reviewer: object is only used once so it can just be a standalone structure under the endpoint

| popout_asset       | string  | The URL of the banner popout image                              |
| revert_text_color? | boolean | Whether to revert the text color on the banner                  |

###### Coachmark Collectibles Marketing Structure

| Field                 | Type    | Description                                                     |
| --------------------- | ------- | --------------------------------------------------------------- |
| type                  | integer | The [collectibles marketing type](#collectibles-marketing-type) |
| title                 | string  | The title of the coachmark                                      |
| body                  | string  | The body text of the coachmark                                  |
| asset_dark            | string  | The URL of the dark mode coachmark image                        |
| asset_light           | string  | The URL of the light mode coachmark image                       |
| version               | integer | The version of the coachmark                                    |
| ref_target_background | object  | The target background configuration                             |

> ###### Collectible Style Structure

Reviewer: why is this in the marketing object instead of the product or category object (rule of thumb is put under most applicable object or the first one it appears in)

| Field             | Type           | Description                   |
| ----------------- | -------------- | ----------------------------- |
| background_colors | array[integer] | An array of background colors |
| button_colors     | array[integer] | An array of button colors     |
| confetti_colors   | array[integer] | An array of confetti colors   |

> ###### Collectible User Discount Structure

Reviewer: why is this in the marketing object (should probably be made its own object or moved under one of the endpoints)

| Field       | Type               | Description               |
| ----------- | ------------------ | ------------------------- |
| amount      | integer            | How much the discount is  |
| discount_id | snowflake          | The discount ID           |
| expires_at  | ?ISO8601 timestamp | When the discount expires |

> ###### Gift Eligibility Structure

Reviewer: why is this in the marketing object (same as above)

| Field | Type    | Description                                           |
| ----- | ------- | ----------------------------------------------------- |
| valid | boolean | Whether the recipient is eligible to receive the gift |

> ###### Variant Return Style

Reviewer: why is this in the marketing object (same as above)

| Value | Name                | Description                                        |
| ----- | ------------------- | -------------------------------------------------- |
| 1     | INDIVIDUAL_PRODUCTS | Variants should be returned as individual products |
| 2     | VARIANTS_GROUP      | Variants should be returned as a group             |

###### Collectibles Shop Home Config

| Value                  | Name                   | Description                                        |
| ---------------------- | ---------------------- | -------------------------------------------------- |
| default                | DEFAULT                | The default shop home configuration                |
| default_with_orb_shelf | DEFAULT_WITH_ORB_SHELF | The default shop home configuration with orb shelf |
| orb_tab                | ORB_TAB                | The orb tab shop home configuration                |
| summer_sale_takeover   | SUMMER_SALE_TAKEOVER   | The summer sale takeover shop home configuration   |

> ###### Collectibles Shop Home Tab

Reviewer: why is this in the marketing object. move under the Get Collectibles Shop endpoint's query params

| Value   | Name    | Description     |
| ------- | ------- | --------------- |
| home    | HOME    | The home tab    |
| catalog | CATALOG | The catalog tab |
| orbs    | ORBS    | The orbs tab    |

###### Shop Block Type

| Value | Name             | Description                    |
| ----- | ---------------- | ------------------------------ |
| 0     | HERO             | A hero shop block              |
| 1     | FEATURED         | A featured shop block          |
| 2     | FEED             | A feed shop block              |
| 3     | WIDE_BANNER      | A wide banner shop block       |
| 4     | SHELF            | A shelf shop block             |
| 5     | COUNTDOWN_TIMER  | A countdown timer shop block   |
| 6     | IMMERSIVE_BANNER | An immersive banner shop block |

###### Shop Item Types

| Value             | Name              | Description                      |
| ----------------- | ----------------- | -------------------------------- |
| AVATAR_DECORATION | Avatar Decoration | An avatar decoration product     |
| PROFILE_EFFECT    | Profile Effect    | A profile effect product         |
| NAMEPLATE         | Nameplate         | A nameplate product              |

> | BUNDLE            | Bundle            | A bundle of collectible products |

Reviewer: ```suggestion
| Value             | Description                      |
| ----------------- | -------------------------------- |
| AVATAR_DECORATION | An avatar decoration product     |
| PROFILE_EFFECT    | A profile effect product         |
| NAMEPLATE         | A nameplate product              |
| BUNDLE            | A bundle of collectible products |
```
also remove name from the below enums

> ###### Shop Product Themes

Reviewer: why is this in the marketing object (same as above)
also applies to all the below enums and structures, i don't wanna copy paste the same review again

| relevance    | RELEVANCE    | Sort by relevance   |
| price        | PRICE        | Sort by price       |
| alphabetical | ALPHABETICAL | Sort alphabetically |
| recency      | RECENCY      | Sort by recency     |
| popularity   | POPULARITY   | Sort by popularity  |

###### Pagination Structure

| Field    | Type    | Description                    |
| -------- | ------- | ------------------------------ |
| offset   | integer | The offset of the current page |
| limit    | integer | The limit of items per page    |
| total    | integer | The total number of items      |
| has_more | boolean | Whether there are more pages   |

###### Collectibles Marketing Type

| Value | Name      | Description |
| ----- | --------- | ----------- |
| 0     | COACHTIP  | A coachtip  |
| 1     | BADGE     | A badge     |
| 2     | BANNER    | A banner    |
| 3     | COACHMARK | A coachmark |

###### Collectibles Marketing Platform Type

| Value | Name    | Description |
| ----- | ------- | ----------- |
| 0     | DESKTOP | Desktop     |
| 1     | MOBILE  | Mobile      |

## Endpoints

<RouteHeader method="GET" url="/collectibles-categories/v2">
  Get Collectibles Categories V2
</RouteHeader>

> Returns the list of [collectible categories](#collectible-category-structure) available in the store.

Reviewer: ```suggestion
Returns the collectible categories available in the store.
```

| include_unpublished?          | boolean | Whether to include unpublished categories                                                      |
| no_cache?                     | boolean | Whether to bypass the cache                                                                    |
| payment_gateway?              | integer | The [payment gateway](/resources/billing#payment-gateway) of the payment source                |
| skip_num_categories?          | integer | The number of categories to skip                                                               |
| variants_return_style?        | integer | The [variant style](#variant-return-style) to return                                           |

###### Response Body

| Field           | Type                                                                            | Description                                         |
| --------------- | ------------------------------------------------------------------------------- | --------------------------------------------------- |
| categories      | array[[collectible category](#collectible-category-structure) object]           | The list of collectible categories                  |

> | user_discounts? | array[[collectible user discount](#collectible-user-discount-structure) object] | The list of user discounts the user is eligible for |

Reviewer: ```suggestion
| categories      | array[[collectible category](#collectible-category-structure) object]           | The collectible categories                  |
| user_discounts? | array[[collectible user discount](#collectible-user-discount-structure) object] | The discounts the user is eligible for |
```
stay concise

| skip_num_categories? ^1^      | integer | The number of categories to skip                                                               |
| variants_return_style?        | integer | The [variant style](#variant-return-style) to return                                           |

^1^ Only usable by Discord employees.

<RouteHeader method="GET" url="/collectibles-shop">
  Get Collectibles Shop
</RouteHeader>

Returns the list of [collectible categories](#collectible-category-structure) available in the store.

###### Query String Params

| Field                         | Type    | Description                                                                                    |
| ----------------------------- | ------- | ---------------------------------------------------------------------------------------------- |
| country_code?                 | string  | The user's [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) country code |
| tab?                          | string  | The [tab](#collectibles-shop-home-tab) to retrieve                                             |

> | shop_home_config?             | string  | The shop home configuration                                                                    |

Reviewer: ```suggestion
| shop_home_config?             | string  | The shop home configuration override                                                                   |
```
also missing enum (`default`, `default_with_orb_shelf`, `orb_tab`)

| payment_gateway?              | integer | The [payment gateway](/resources/billing#payment-gateway) of the payment source                |
| skip_num_categories? ^1^      | integer | The number of categories to skip                                                               |
| variants_return_style?        | integer | The [variant style](#variant-return-style) to return                                           |

^1^ Only usable by Discord employees.

###### Response Body

| Field           | Type                                                                            | Description                                         |
| --------------- | ------------------------------------------------------------------------------- | --------------------------------------------------- |
| categories      | array[[collectible category](#collectible-category-structure) object]           | The list of collectible categories                  |
| shop_blocks     | array[[shop block](#shop-block-object) object]                                  | The list of shop blocks                             |

> | user_discounts? | array[[collectible user discount](#collectible-user-discount-structure) object] | The list of user discounts the user is eligible for |

Reviewer: ```suggestion
| categories      | array[[collectible category](#collectible-category-structure) object]           | The collectible categories                  |
| shop_blocks     | array[[shop block](#shop-block-object) object]                                  | The shop blocks                             |
| user_discounts? | array[[collectible user discount](#collectible-user-discount-structure) object] | The discounts the user is eligible for |
```
stay concise

| pagination | [pagination](#pagination-structure) object | The pagination information   |
| skus       | array[snowflake]                           | The list of matching SKU IDs |

<RouteHeader method="GET" url="/collectibles-products/{sku.id}" unauthenticated>
  Get Collectibles Product
</RouteHeader>

Returns the [collectible product](#collectible-product-structure) for a given SKU ID.

###### Query String Params

| Field                  | Type    | Description                                                                                    |
| ---------------------- | ------- | ---------------------------------------------------------------------------------------------- |
| country_code?          | string  | The user's [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) country code |
| include_bundles?       | boolean | Whether to include bundles                                                                     |
| variants_return_style? | integer | The [variant style](#variant-return-style) to return                                           |

<RouteHeader method="GET" url="/users/@me/collectibles-purchases">

>   Get Users Purchased Collectibles

Reviewer: ```suggestion
  Get User Purchased Collectibles
```

| --------------------- | ------- | ---------------------------------------------------- |
| variant_return_style? | integer | The [variant style](#variant-return-style) to return |

<RouteHeader method="GET" url="/user-profile-effects" unauthenticated>
  Get User Profile Effects
</RouteHeader>

Returns the list of [profile effects](https://support.discord.com/hc/en-us/articles/17828465914263-Profile-Effects) available to display.

###### Query String Params

| Field                 | Type    | Description                                    |
| --------------------- | ------- | ---------------------------------------------- |
| locale?               | string  | The user's locale                              |
| with_unpublished? ^1^ | boolean | Whether to include unpublished profile effects |

^1^ Only usable by Discord employees.

###### Response Body

| Field                  | Type                                                                              | Description                                  |
| ---------------------- | --------------------------------------------------------------------------------- | -------------------------------------------- |

> | profile_effect_configs | array[[profile effect config](#profile-effect-collectible-item-structure) object] | The list of profile effects available to use |

Reviewer: ```suggestion
| profile_effect_configs | array[[profile effect collectible item](#profile-effect-collectible-item-structure) object] | The profile effects available to display |
```

  "animationType": 2,
  "thumbnailPreviewSrc": "https://cdn.discordapp.com/assets/profile_effects/effects/b17d139f2e9/splash/thumbnail.png",
  "reducedMotionSrc": "https://cdn.discordapp.com/assets/profile_effects/effects/b17d139f2e9/splash/reducedMotion.png",
  "effects": [
    {
      "src": "https://cdn.discordapp.com/assets/profile_effects/effects/b17d139f2e9/splash/intro.png",
      "loop": false,
      "height": 880,
      "width": 450,
      "duration": 2880,
      "start": 0,
      "loopDelay": 0,
      "position": {
        "x": 0,
        "y": 0
      },
      "zIndex": 100,
      "randomizedSources": []
    },
    {
      "src": "https://cdn.discordapp.com/assets/profile_effects/effects/b17d139f2e9/splash/loop.png",
      "loop": true,
      "height": 880,
      "width": 450,
      "duration": 2880,
      "start": 5760,
      "loopDelay": 5760,
      "position": {
        "x": 0,
        "y": 0
      },
      "zIndex": 100,
      "randomizedSources": []
    }
  ]
}
```

<RouteHeader method="GET" url="/users/@me/valid-collectibles-gift-recipient">
  Get Valid Collectibles Gift Recipient
</RouteHeader>

> Returns [gift eligibility](#gift-eligibility-structure) object for a given user for given collectible SKU ID.

Reviewer: ```suggestion
Returns a [gift eligibility](#gift-eligibility-structure) object for the given user and collectible SKU ID.
```

###### Query String Params

| Field        | Type      | Description                                      |
| ------------ | --------- | ------------------------------------------------ |
| recipient_id | snowflake | The ID of the user to check for gift eligibility |
| sku_id       | snowflake | The SKU ID of the collectible to gift            |

<RouteHeader method="GET" url="/users/@me/valid-collectibles-gift-recipients-batch">
  Get Valid Collectibles Gift Recipients Batch
</RouteHeader>

> Returns a mapping of SKU IDs to [gift eligibility](#gift-eligibility-structure) object for a given user for given collectible SKU IDs.

Reviewer: ```suggestion
Returns a mapping of SKU IDs to [gift eligibility](#gift-eligibility-structure) object for the given user and collectible SKU IDs.
```

###### Query String Params

| Field        | Type             | Description                                      |
| ------------ | ---------------- | ------------------------------------------------ |
| recipient_id | snowflake        | The ID of the user to check for gift eligibility |
| skus_id      | array[snowflake] | The SKU IDs of the collectibles to gift          |

<RouteHeader method="PUT" url="/users/@me/claim-premium-collectibles-product">
  Claim Premium Collectibles Product
</RouteHeader>

> Returns list of [collectible products](#collectible-product-structure) objects owned by the current user if the user hasn't claimed the product yet, otherwise returns 204 empty response.

Reviewer: ```suggestion
Claims a collectible SKU provided for free to premium users. Returns a list of [collectible products](#collectible-product-structure) objects owned by the current user if the user hasn't claimed the product yet, otherwise returns a 204 empty response.
```
you didn't specify what the endpoint actually does

###### JSON Params

| Field  | Type      | Description                                   |
| ------ | --------- | --------------------------------------------- |

> | sku_id | snowflake | The SKU ID of the premium collectible product |

Reviewer: ```suggestion
| sku_id | snowflake | The SKU ID of the premium collectible product to claim |
```

<RouteHeader method="GET" url="/users/@me/collectibles-marketing">
  Get Collectibles Marketing
</RouteHeader>

> Returns the [collectibles marketing](#collectibles-marketing-object) information for the current user.

Reviewer: ```suggestion
Returns collectibles marketing information for the current user.
```

| google_sku_ids?        | ?map[integer, string]                                                         | The Google SKU IDs for the collectible per [purchase type](/resources/store#subscription-plan-purchase-type) |
| items ^3^              | array[[collectible item](#collectible-item-object) object]                    | The items included in the collectible product                                                                |
| name                   | string                                                                        | The name of the collectible product                                                                          |
| name_localizations?    | ?object                                                                       | The localized names of the collectible product                                                               |
| premium_type           | integer                                                                       | The premium type required to purchase the collectible product                                                |

> | prices?                | map[integer, [prices](/resources/store#subscription-prices-structure) object] | The prices for the collectible per [purchase type](/resources/store#subscription-plan-purchase-type)         |

Reviewer 2: ```suggestion
| prices?                | map[integer, [subscription prices](/resources/store#subscription-prices-structure) object] | The prices for the collectible per [purchase type](/resources/store#subscription-plan-purchase-type)         |
```

| banner_animated_url | string                                         | The URL of the animated immersive banner image |

### Collectibles Marketing Object

###### Coachtip Collectibles Marketing Structure

| Field                 | Type          | Description                                                     |
| --------------------- | ------------- | --------------------------------------------------------------- |
| type                  | integer       | The [collectibles marketing type](#collectibles-marketing-type) |
| version               | integer       | The version of the coachtip                                     |
| title                 | string        | The title of the coachtip                                       |
| body                  | string        | The body text of the coachtip                                   |
| avatar                | string        | The URL of the coachtip avatar                                  |
| decorations           | array[string] | The avatar decoration asset hashes                              |
| dismissible_content   | integer       | The ID of the dismissible content                               |

> | ref_target_background | object        | The target background configuration                             |

Reviewer 2: isnt this supposed to be target-background-structure

###### Target Background Structure

| Field | Type   | Description                               |
| ----- | ------ | ----------------------------------------- |
| style | object | The style configuration of the background |
| asset | object | The asset configuration of the background |

###### Target Background Information Structure

| Field   | Type   | Description                     |
| ------- | ------ | ------------------------------- |
| resting | object | The resting state configuration |

> | hovered | object | The hovered state configuration |

Reviewer 2: what are actual types instead of just `object`

| include_bundles?              | boolean | Whether to include bundles                                                                     |
| include_dynamic_blocks?       | boolean | Whether to include dynamic blocks                                                              |
| include_nameplates_on_mobile? | boolean | Whether to include nameplates on mobile                                                        |
| include_unpublished? ^1^      | boolean | Whether to include unpublished categories                                                      |
| no_cache? ^1^                 | boolean | Whether to bypass the cache                                                                    |
| payment_gateway?              | integer | The [payment gateway](/resources/billing#payment-gateway) of the payment source                |
| skip_num_categories? ^1^      | integer | The number of categories to skip                                                               |
| variants_return_style?        | integer | The [variant style](#variant-return-style) to return                                           |

> ^1^ Only usable by Discord Employees.

Reviewer 2: ```suggestion
^1^ Only usable by Discord employees.
```

| shop_blocks     | array[[shop blocks](#shop-block-object) object]                                 | The list of shop blocks                             |
| user_discounts? | array[[collectible user discount](#collectible-user-discount-structure) object] | The list of user discounts the user is eligible for |

<RouteHeader method="GET" url="/shop/search">
  Search Collectible Products
</RouteHeader>

Returns the list of SKU IDs that match the given query.

###### Query String Params

| Field           | Type          | Description                                                                 |
| --------------- | ------------- | --------------------------------------------------------------------------- |
| item_types?     | array[string] | The [item types](#shop-item-types) to search for                            |
| colors?         | array[string] | The products to search for that have the [color](#shop-product-colors)      |
| themes?         | array[string] | The products with the specified [theme](#shop-product-themes) to search for |
| orbs_eligible?  | boolean       | Whether you can purchase the products with orbs                             |

> | offset?         | int           | Number of products to skip before returning results                         |

Reviewer 2: ```suggestion
| offset?         | integer       | Number of products to skip before returning results                         |
```

| recipient_id | snowflake        | The ID of the user to check for gift eligibility |
| skus_id      | array[snowflake] | The SKU IDs of the collectibles to gift          |

<RouteHeader method="PUT" url="/users/@me/claim-premium-collectibles-product">
  Claim Premium Collectibles Product
</RouteHeader>

Returns list of [collectible products](#collectible-product-structure) objects owned by the current user if the user hasn't claimed the product yet, otherwise returns 204 empty response.

###### JSON Params

| Field  | Type      | Description                                   |
| ------ | --------- | --------------------------------------------- |
| sku_id | snowflake | The SKU ID of the premium collectible product |

<RouteHeader method="GET" url="/users/@me/collectibles-marketing">
  Get Collectibles Marketing
</RouteHeader>

Returns the [collectibles marketing](#collectibles-marketing-object) information for the current user.

###### Query String Params

| Field     | Type | Description                                                                        |
| --------- | ---- | ---------------------------------------------------------------------------------- |

> | platform? | int  | The [platform](#collectibles-marketing-platform-type) to get marketing information for |

Reviewer 2: ```suggestion
| platform? | integer | The [platform](#collectibles-marketing-platform-type) to get marketing information for |
```
